docs/webapp.adoc

```diff
@@ -1,30 +1,625 @@
 
 {zwsp} +
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+{zwsp} +
+{zwsp} +
 {zwsp} +
+
+
+== 1. Completely standalone
+
+In this first of the two approaches, we'll minimize the use of runtime XP: all values and asset URLs are hardcoded. XP is used to serve the initial HTML and the assets, as well as providing a guillotine endpoint where the browser can fetch data, but this functionality can easily be replaced with any other file- and data-serving solution and still work just fine.
+
+[[html1]]
+=== HTML base
+
+The webapp begins with some basic HTML, setting it all up in the browser.
+
+.webapp.html:
+[source,html,options="nowrap"]
+----
+<!DOCTYPE html>
+<html>
+    <head>
+        <meta charset="UTF-8">
+        <title>Completely standalone</title>
+
+
+        <!--1-->
+        <script crossorigin src="https://unpkg.com/react@16/umd/react.production.min.js"></script>
+        <script crossorigin src="https://unpkg.com/react-dom@16/umd/react-dom.production.min.js"></script>
+
+        <!--2-->
+        <script src="(...) /react4xp/shared.d366a6c1b.js"></script>
+        <link rel="stylesheet" type="text/css" href="(...) /react4xp/shared.5440dda80.css" />
+        <link rel="stylesheet" type="text/css" href="(...) /react4xp/MovieList.css" />
+
+        <!--3-->
+        <script src="(...) /react4xp/MovieList.js"></script>
+
+        <style>
+            body { margin: 0; padding: 0; }
+            h1, p, .faux-spinner { padding: 30px; margin: 0 auto; font-family: 'DejaVu Sans', Arial, Helvetica, sans-serif; }
+        </style>
+    </head>
+
+    <body>
+        <h1>Top 3 movies to put in a list</h1>
+        <p>#4 will blow your mind!</p>
+
+        <!--4-->
+        <div id="movieListContainer">
+            <div class="faux-spinner">Loading movies...</div>
+        </div>
+
+        <!--5-->
+        <script>
+            var MOVIE_LIST_PARAMS= {
+                parentPath: '/moviesite',
+                apiUrl: '/admin/site/preview/default/draft/moviesite/api/headless',
+                movieType: 'com.enonic.app.react4xp:movie',
+                movieCount: 3,
+                sortExpression: 'data.year ASC',
+            };
+        </script>
+
+        <!--6-->
+        <script defer src="(...) /webapp/script.js"></script>
+    </body>
+</html>
+
+----
+<1> We start by running React and ReactDOM from a CDN.
+<2> Next, we fetch 3 dependency chunks that the _MovieList_ entry needs: _shared.&lt;hash&gt;.js_, _shared.&lt;hash&gt;.css_ and _MovieList.css_.
++
+[NOTE]
+====
+As before, these are asset URLs that depend on your setup - and this applies to both the *content-dependent &lt;hash&gt;* in the filename and the *asset path* itself. On my local machine running XP, the path (the `(...)` before _/react4xp/..._) is this:
+
+`/webapp/com.enonic.app.react4xp/_/asset/com.enonic.app.react4xp:1604314030`
```

Comment:
`/_/` should be globally available so no need for the /webapp/com.enonic.app.react4xp/_/

Reply:
👍

Reply:
✅